thentos-purescript/src/Register.purs

```diff
@@ -0,0 +1,207 @@
+        ]
+
+    -- can not register: already logged in
+    Tuple false true -> H.div [cl "login-success"] [H.p_ [translate "TR__REGISTRATION_ALREADY_LOGGED_IN"]]
+
+    -- registered: waiting for processing of activation email
+    Tuple true false -> H.div [cl "login-success"]
+        [ H.h2_ [translate "TR__REGISTER_SUCCESS"]
+        , H.p_ [translate "TR__REGISTRATION_CALL_FOR_ACTIVATION"]
+
+        -- FIXME: the a3 code sais this.  what does it mean?:
+        -- 'Show option in case the user is not automatically logged in (e.g. 3rd party cookies blocked.)'
+        ]
+
+    -- FIXME: a3 code sais this.  is that relevant for us?
+    -- <!-- FIXME: Technically this should only display if you logged in as the user you just registered as, but
+    -- this will display if you log in as any user -->
+
+    -- registered and registration link clicked
+    Tuple true true -> H.div [cl "login-success"]
+        [ H.h2_ [translate "TR__REGISTRATION_THANKS_FOR_REGISTERING"]
+        , H.p_ [translate "TR__REGISTRATION_PROCEED"]  -- FIXME: link
+        ]
+
+
+mydebug :: State -> ComponentHTML Query
+mydebug st = H.pre_
+    [ H.p_ [H.text $ show st.stErrors]
+    , H.p_ [H.text $ show st.stLoggedIn]
+    , H.p_ [H.text $ show st.stRegSuccess]
+    , H.p_ [H.text $ show st.stName]
+    , H.p_ [H.text $ show st.stEmail]
+    , H.p_ [H.text $ show st.stPass1]
+    , H.p_ [H.text $ show st.stPass2]
+    , H.p_ [H.text $ show st.stTermsAndConds]
+    ]
+
+errors :: State -> Array (ComponentHTML Query)
+errors st = f <$> st.stErrors
+  where
+    f :: String -> ComponentHTML Query
+    f msg = H.div [cl "form-error"] [H.p_ [translate msg]]
+
+inputField :: P.InputType -> ComponentHTML Query -> String
+           -> (String -> State -> State)
+           -> ComponentHTML Query
+inputField inputType msg key updateState = H.label_
+    [ H.span [cl "label-text"] [msg]
+    , H.input [ P.inputType inputType, P.name key, P.required true
+              , E.onInput $ E.input $ KeyPressed <<< updateState <<< onChangeValue
+              ]
+    ]
+
+-- there is something about this very similar to `translate`: we want to be able to collect all
+-- classnames occurring in a piece of code, and construct a list from them with documentation
+-- (source file locations?).
+cl :: forall r i. String -> P.IProp (class :: P.I | r) i
+cl = P.class_ <<< H.className
+
+
+-- * eval
+
+eval :: forall g. Natural Query (ComponentDSL State Query g)
+eval (KeyPressed updateState next) = do
+    modify updateState
+    modify checkState
+    pure next
+eval (UpdateTermsAndConds newVal next) = do
+    modify (\st -> st { stTermsAndConds = newVal })
+    modify checkState
+    pure next
+
+checkState :: State -> State
+checkState st = st { stErrors = passwordMismatch st ++ invalidEmail st }
+  where
+    passwordMismatch :: State -> Array String
+    passwordMismatch st = if st.stPass1 == st.stPass2 then [] else ["passwords must match"]
```

Comment:
(Nitpicking, feel free to ignore) Why not make `passwordMismatch` just an `Array String` and use the `st` from the outer scope?